lib/distrobaker.py

```diff
@@ -766,16 +794,25 @@ def get_buildsys(which):
     if 'main' not in c:
         logger.critical('DistroBaker is not configured, aborting.')
         return None
-    if which != 'source' and which != 'destination':
+    if which not in ('source', 'destination'):
         logger.error('Cannot get "%s" build system.', which)
         return None
-    if not hasattr(get_buildsys, which):
+
+    session_timed_out = False
+    if hasattr(get_buildsys, which):
+        session_age = datetime.datetime.now() - getattr(get_buildsys, which)['session_start_time']
+        #slightly less than an hour, to be safe
+        if session_age.seconds > 3550 or session_age.days > 0:
```

Comment:
Not clear why do you need `session_age.days > 0`
if `session_age.days > 0`  then `session_age.seconds > 3550` and we don't need to compare it with days, right?

Reply:
Actually no, because the difference between the two times is stored as a `datetime.timedelta` object which stores only days and seconds (not hours or minutes) as a sum. When the delta gets longer than one day, it sets days to 1 and zeroes the seconds, so if the session is longer than one day, but shorter than one day and one hour, the seconds will be between 0 and 3600.